test/gtest/common/test_helpers.h

```diff
@@ -257,20 +257,32 @@ class ptr_vector<void> : public ptr_vector_base<void> {
 /**
  * Safely wraps C handles
  */
-template <typename T>
+template <typename T, typename ArgT = void *>
 class handle {
 public:
     typedef T handle_type;
     typedef void (*dtor_t)(T handle);
+    typedef void (*dtor2_t)(T handle, ArgT arg);
```

Comment:
FYI a "C++" approach would be to use something like std::bind to create a callable object and pass it as a no-argument destructor - instead of modifying ucs::handle

Reply:
good point but `std::bind` was introduced in c++11, gtests building doesn't support it.